src/components/Table/BaseTable.tsx

```diff
@@ -51,127 +55,171 @@ type SpruceTableProps = {
+                            onConfirm={(value) =>
+                              header.column.setFilterValue(value)
+                            }
+                            options={
+                              meta.treeSelect?.filterOptions
+                                ? meta.treeSelect.options.filter(
+                                    ({ value }) =>
+                                      !!header.column
+                                        .getFacetedUniqueValues()
+                                        .get(value)
+                                  )
+                                : meta.treeSelect.options
+                            }
+                            value={
+                              (header?.column?.getFilterValue() as string[]) ??
+                              []
+                            }
+                          />
+                        ) : (
+                          <TableSearchPopover
+                            data-cy={meta?.search?.["data-cy"]}
+                            onConfirm={(value) =>
+                              header.column.setFilterValue(value)
+                            }
+                            placeholder={meta?.search?.placeholder}
+                            value={
+                              (header?.column?.getFilterValue() as string) ?? ""
+                            }
+                          />
+                        ))}
+                    </HeaderCell>
+                  );
+                })}
+              </HeaderRow>
             ))}
-            {row.original.renderExpandedContent && (
-              <StyledExpandedContent row={row} />
+          </TableHead>
+          <TableBody>
+            {loading && (
+              <TableLoader
+                numColumns={table.getAllColumns().length}
+                numRows={loadingRows}
+              />
             )}
-            {row.subRows &&
-              row.subRows.map((subRow) => (
-                <Row
-                  key={subRow.id}
-                  row={subRow}
-                  className={css`
-                    &[aria-hidden="false"] td > div[data-state="entered"] {
-                      max-height: unset;
-                    }
-                  `}
-                >
-                  {subRow.getVisibleCells().map((cell) => (
-                    <Cell key={cell.id}>
-                      {flexRender(
-                        cell.column.columnDef.cell,
-                        cell.getContext()
-                      )}
-                    </Cell>
-                  ))}
-                </Row>
-              ))}
-          </Row>
+            <RenderRows rows={rows} virtualRows={virtualRows} />
+          </TableBody>
+        </StyledTable>
+        {!loading &&
+          rows.length === 0 &&
+          (emptyComponent || "No data to display")}
+      </>
+    );
+  }
+);
+
+const RenderRows = <T extends LGRowData>({
```

Comment:
[nit] Slight preference to include this as a ternary in BaseTable instead of its own component, but your call.